src/mdio/converters/segy.py

```diff
@@ -192,6 +169,19 @@ def _scan_for_headers(
     return segy_dimensions, segy_headers
 
 
+def _read_segy_file_pp(segy_kw: SegyFileArguments) -> tuple[int, np.NDArray[np.int32], SegyFileHeaderDump]:
```

Comment:
please rename this to be something that explains its purpose. for inspiration: `_read_segy_file_info`.